src/Mutation/Collection.php

```diff
@@ -2,18 +2,20 @@
 
 namespace Softonic\GraphQL\Mutation;
 
+use Softonic\GraphQL\Exceptions\InaccessibleArgumentException;
+
 class Collection extends FilteredCollection
 {
-    public function __get(string $key)
+    public function __get(string $key): Collection
     {
         if (empty($this->arguments)) {
-            throw new \Exception('You cannot access a non existing collection');
+            throw new InaccessibleArgumentException('You cannot access a non existing collection');
```

Comment:
It would be nice to have a `throw InaccessibleArgumentException::fromEmptyArguments(...)`